devicetypes/smartthings/zigbee-sound-sensor/zigbee-sound-sensor.groovy

```diff
@@ -0,0 +1,173 @@
+        map = getBatteryPercentageResult(Integer.parseInt(descMap.value, 16))
+    } else if(descMap?.clusterInt == zigbee.TEMPERATURE_MEASUREMENT_CLUSTER && descMap.commandInt == 0x07) {
+        if (descMap.data[0] == "00") {
+            sendEvent(name: "checkInterval", value: 60 * 60, displayed: true, data: [protocol: "zigbee", hubHardwareId: device.hub.hardwareID])
+        } else {
+            log.warn "TEMP REPORTING CONFIG FAILED - error code: ${descMap.data[0]}"
+        }
+    } else if(descMap.clusterInt == 0x0020 && descMap.commandInt == 0x00) {
+        log.debug "Check in interval command received!"
+        sendEvent(name: "checkInterval", value: 60 * 60, displayed: true, data: [protocol: "zigbee", hubHardwareId: device.hub.hardwareID])
+    } else if (map.name == "temperature") {
+        if (tempOffset) {
+            map.value = (int) map.value + (int) tempOffset
+        }
+        map.descriptionText = temperatureScale == 'C' ? '${device.displayName} was ${value}°C' : '${device.displayName} was ${value}°F'
+        map.translatable = true
+    }
+
+    return map
+}
+
+private Map getBatteryPercentageResult(rawValue) {
+    def result = [:]
+    def volts = rawValue / 10
+    if (!(rawValue == 0 || rawValue == 255)) {
+        def minVolts = 2.2
+        def maxVolts = 3.0
+        def pct = (volts - minVolts) / (maxVolts - minVolts)
+        def roundedPct = Math.round(pct * 100)
+        if (roundedPct <= 0)
+            roundedPct = 1
+        result.value = Math.min(100, roundedPct)
+    }
+    log.debug "Battery Percentage rawValue = ${rawValue} -> ${(rawValue-23)/7.0 * 100}"
+    result.name = 'battery'
+    result.translatable = true
+    result.descriptionText = "${device.displayName} battery was ${result.value}%"
+    return result
+}
+
+private Map getSoundDetectionResult(value) {
+    def text = "Sound was ${value}"
+    def result = [name: "sound", value: value, descriptionText: text, displayed: true]
+    return result
+}
+
+def ping() {
+    log.debug "Ping call"
+    return zigbee.readAttribute(zigbee.POWER_CONFIGURATION_CLUSTER, 0x0020) +
+            zigbee.readAttribute(zigbee.TEMPERATURE_MEASUREMENT_CLUSTER, 0x0000)
+}
+
+def refresh() {
+    log.debug "Refresh call"
+    return zigbee.readAttribute(zigbee.POWER_CONFIGURATION_CLUSTER, 0x0020) +
+            zigbee.readAttribute(zigbee.TEMPERATURE_MEASUREMENT_CLUSTER, 0x0000)
+}
+
+def configure() {
+    log.debug "Configure Function Called"
+
+    sendEvent(name: "checkInterval", value: 60 * 60, displayed: false, data: [protocol: "zigbee", hubHardwareId: device.hub.hardwareID])
+
+    //send zone enroll response, configure short and long poll, fast poll timeout and check in interval
+    def enrollCmds = (zigbee.command(0x0020, 0x02, "B0040000") + zigbee.command(0x0020, 0x03, "0200") +
+            zigbee.writeAttribute(0x0020, 0x0003, 0x21, 0x0028) + zigbee.writeAttribute(0x0020, 0x0000, 0x23, 0x00001950))
```

Comment:
can we replace any of these cluster numbers with the static values from `zigbee` for code readability?

Reply:
Most of those number can be covered, except poll control cluster (0x0020) as it isn't specified as static value in zigbee source code, but I can provide some get methods for it.